src/big/big_decimal.cr

```diff
@@ -189,6 +189,22 @@ struct BigDecimal < Number
     self * BigDecimal.new(other)
   end
 
+  def %(other : BigDecimal) : BigDecimal
+    if @scale > other.scale
+      scaled = other.scale_to(self)
+      BigDecimal.new(@value % scaled.value, @scale)
+    elsif @scale < other.scale
+      scaled = scale_to(other)
+      BigDecimal.new(scaled.value % other.value, other.scale)
+    else
+      BigDecimal.new(@value % other.value, @scale)
+    end
+  end
+
+  def %(other : Int)
```

Comment:
Same overload could be added for `Float` perhaps?

Reply:
Yeah, I'll add that. Thanks!